bug android template

**Describe the bug**
cargo mobile init missing too many file

**Steps To Reproduce**
cargo mobile init

fund only a  MainActivity.kt file in my app folder 


**Screenshots**
![QQ_1723805152390](https://github.com/user-attachments/assets/6e07a46d-18f0-4030-8dec-cce487333583)
![QQ_1723805213167](https://github.com/user-attachments/assets/31556cc6-ddab-4e03-a8c8-1a39950dfc0f)

```
cargo android run
 (V2243A) with target "aarch64-linux-android"
   Compiling zerocopy-derive v0.7.35
   Compiling serde_derive v1.0.208
   Compiling cssparser v0.27.2
   Compiling thiserror-impl v1.0.63
   Compiling cssparser-macros v0.6.1
   Compiling html5ever v0.26.0
   Compiling derive_more v0.99.18
   Compiling regex v1.10.6
   Compiling tao-macros v0.1.2
   Compiling anyhow v1.0.86
   Compiling paste v1.0.15
   Compiling http v0.2.12
   Compiling sha2 v0.10.8TING [4s]
   Compiling url v2.5.2CUTING [4s]
   Compiling env_logger v0.10.25s]
   Compiling wry v0.37.0UTING [6s]
error: failed to run custom build command for `wry v0.37.0`

Caused by:
  process didn't exit successfully: `D:\tauri\a\target\debug\build\wry-a64b02ec26971436\build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-env-changed=WRY_ANDROID_PACKAGE
  cargo:rerun-if-env-changed=WRY_ANDROID_LIBRARY
  cargo:rerun-if-env-changed=WRY_ANDROID_KOTLIN_FILES_OUT_DIR

  --- stderr
  thread 'main' panicked at C:\Users\Administrator\.cargo\registry\src\index.crates.io-6f17d22bba15001f\wry-0.37.0\build.rs:37:11:
  Failed to canonicalize `WRY_ANDROID_KOTLIN_FILES_OUT_DIR` path D:\tauri\a\gen\android/app/src/main/kotlin/com/example
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: Failed to compile lib
    `Failed to run `cargo build`: command ["cargo", "build", "--package",
    "a", "--manifest-path", "D:\\tauri\\a\\Cargo.toml", "--target",
    "aarch64-linux-android"] exited with code 101

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:rustBuildArm64Debug'.
> Process 'command 'cargo'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.7/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 15s
error: Failed to assemble APK
    command ["D:\\tauri\\a\\gen\\android\\gradlew.bat", "--project-dir", "D:\\tauri\\a\\gen\\android"] exited with code
```
```
 cargo mobile doctor

[✔] cargo-mobile v0.13.2
    • Contains commits up to "feat: Add a Dioxus template (#357)\n"
    • Installed at "C:\\Users\\Administrator\\.cargo\\.cargo-mobile2"
    • Windows 11 v10.0
    • rustc v1.80.1 (3f5fd8dd4 2024-8-6)

[✔] Android developer tools
    • SDK v26.1.1 installed at "D:\\Android\\sdk"
    • NDK v25.0.8775105 installed at "D:\\Android\\sdk\\ndk\\25.0.8775105"

[✔] Connected devices
 (V2243A)ia 5230
```

help~

--- a/.changes/App-identifier.md
+++ b/.changes/App-identifier.md
@@ -2,4 +2,4 @@
 "cargo-mobile2": "minor"
 ---
 
-Removed `App::reverse_identifier` and Added `App::identifier`
+The app identifier must now be provided in reverse order (e.g. `dev.tauri.app` instead of `app.tauri.dev`). Removed `App::reverse_identifier` and Added `App::identifier`.

--- a/.changes/App-identifier.md
+++ b/.changes/App-identifier.md
@@ -1,0 +1,5 @@
+---
+"cargo-mobile2": "minor"
+---
+
+Removed `App::reverse_identifier` and Added `App::identifier`

--- a/src/android/config.rs
+++ b/src/android/config.rs
@@ -203,7 +203,7 @@ impl Config {
     pub fn from_raw(app: App, raw: Option<Raw>) -> Result<Self, Error> {
         let raw = raw.unwrap_or_default();
 
-        if app.reverse_identifier().contains('-') {
+        if app.identifier().contains('-') {
             return Err(Error::IdentifierCannotContainHyphens);
         }
 

--- a/src/android/device.rs
+++ b/src/android/device.rs
@@ -363,7 +363,7 @@ impl<'a> Device<'a> {
             self.install_apk(config, env, profile)
                 .map_err(RunError::ApkInstallFailed)?;
         }
-        let activity = format!("{}/{}", config.app().reverse_identifier(), activity);
+        let activity = format!("{}/{}", config.app().identifier(), activity);
         self.adb(env)
             .before_spawn(move |cmd| {
                 cmd.args(["shell", "am", "start", "-n", &activity]);
@@ -390,7 +390,7 @@ impl<'a> Device<'a> {
         let stdout = loop {
             let cmd = duct::cmd(
                 env.platform_tools_path().join("adb"),
-                ["shell", "pidof", "-s", &config.app().reverse_identifier()],
+                ["shell", "pidof", "-s", &config.app().identifier()],
             )
             .vars(env.explicit_env())
             .stderr_capture()

--- a/src/apple/config/mod.rs
+++ b/src/apple/config/mod.rs
@@ -297,7 +297,7 @@ pub struct Config {
 
 impl Config {
     pub fn from_raw(app: App, raw: Option<Raw>) -> Result<Self, Error> {
-        if app.reverse_identifier().contains('_') {
+        if app.identifier().contains('_') {
             return Err(Error::IdentifierCannotContainUnderscores);
         }
 

--- a/src/apple/device/simctl/run.rs
+++ b/src/apple/device/simctl/run.rs
@@ -45,7 +45,7 @@ pub fn run(
 
     handle.wait().map_err(RunError::DeployFailed)?;
 
-    let app_id = config.app().reverse_identifier();
+    let app_id = config.app().identifier();
     let mut launcher_cmd = duct::cmd("xcrun", ["simctl", "launch", id, &app_id])
         .vars(env.explicit_env())
         .dup_stdio();

--- a/src/config/app/mod.rs
+++ b/src/config/app/mod.rs
@@ -213,17 +213,12 @@ impl App {
         &self.stylized_name
     }
 
-    pub fn reverse_identifier(&self) -> String {
-        self.identifier
-            .clone()
-            .split('.')
-            .rev()
-            .collect::<Vec<_>>()
-            .join(".")
+    pub fn identifier(&self) -> &str {
+        &self.identifier
     }
 
     pub fn android_identifier_escape_kotlin_keyword(&self) -> String {
-        self.reverse_identifier()
+        self.identifier
             .split('.')
             .map(|s| {
                 if crate::reserved_names::KOTLIN_ONLY_KEYWORDS.contains(&s) {

--- a/src/config/app/raw.rs
+++ b/src/config/app/raw.rs
@@ -19,27 +19,32 @@ enum DefaultIdentifierError {
     FailedToParseEmailAddr,
 }
 
-fn default_identifier(_wrapper: &TextWrapper) -> Result<Option<String>, DefaultIdentifierError> {
+fn default_identifier(
+    _wrapper: &TextWrapper,
+    name: &str,
+) -> Result<Option<String>, DefaultIdentifierError> {
     let email = Git::new(".".as_ref())
         .user_email()
         .map_err(DefaultIdentifierError::FailedToGetGitEmailAddr)?;
-    let identifier = email
+    let domain = email
         .trim()
         .split('@')
         .last()
         .ok_or(DefaultIdentifierError::FailedToParseEmailAddr)?;
     Ok(
-        if !COMMON_EMAIL_PROVIDERS.contains(&identifier)
-            && identifier::check_identifier_syntax(identifier).is_ok()
+        if !COMMON_EMAIL_PROVIDERS.contains(&domain)
+            && identifier::check_identifier_syntax(domain).is_ok()
         {
             #[cfg(not(feature = "brainium"))]
-            if identifier == "brainiumstudios.com" {
+            if domain == "brainiumstudios.com" {
                 crate::util::cli::Report::action_request(
                     "You have a Brainium email address, but you're using a non-Brainium installation of cargo-mobile2!",
                     "If that's not intentional, run `cargo install --force --git https://github.com/tauri-apps/cargo-mobile2 --features brainium`",
                 ).print(_wrapper);
             }
-            Some(identifier.to_owned())
+
+            let reverse_domain = domain.split('.').rev().collect::<Vec<_>>().join(".");
+            Some(format!("{reverse_domain}{name}"))
         } else {
             None
         },
@@ -87,13 +92,15 @@ impl Defaults {
         let dir_name = dir_name
             .to_str()
             .ok_or_else(|| DefaultsError::CurrentDirInvalidUtf8(cwd.clone()))?;
+        let name = name::transliterate(&dir_name.to_kebab_case());
+        let dot_name = name.as_ref().map(|n| format!(".{n}")).unwrap_or_default();
         Ok(Self {
-            name: name::transliterate(&dir_name.to_kebab_case()),
-            stylized_name: dir_name.to_title_case(),
-            identifier: default_identifier(wrapper)
+            identifier: default_identifier(wrapper, &dot_name)
                 .ok()
                 .flatten()
-                .unwrap_or_else(|| "example.com".to_owned()),
+                .unwrap_or_else(|| format!("com.example{dot_name}")),
+            name,
+            stylized_name: dir_name.to_title_case(),
         })
     }
 }

--- a/src/init.rs
+++ b/src/init.rs
@@ -149,19 +149,6 @@ pub fn exec(
     }
     let mut dot_cargo =
         dot_cargo::DotCargo::load(config.app()).map_err(Error::DotCargoLoadFailed)?;
-    // Mysteriously, builds that don't specify `--target` seem to fight over
-    // the build cache with builds that use `--target`! This means that
-    // alternating between i.e. `cargo run` and `cargo apple run` would
-    // result in clean builds being made each time you switched... which is
-    // pretty nightmarish. Specifying `build.target` in `.cargo/config`
-    // fortunately has the same effect as specifying `--target`, so now we can
-    // `cargo run` with peace of mind!
-    //
-    // This behavior could be explained here:
-    // https://doc.rust-lang.org/cargo/reference/config.html#buildrustflags
-    dot_cargo.set_default_target(
-        util::host_target_triple().map_err(Error::HostTargetTripleDetectionFailed)?,
-    );
 
     let metadata = Metadata::load(config.app().root_dir()).map_err(Error::MetadataFailed)?;
 

--- a/src/templating/init.rs
+++ b/src/templating/init.rs
@@ -100,27 +100,27 @@ fn snake_case(
         .map_err(Into::into)
 }
 
-fn reverse_domain(
+fn ident_last_part(
     helper: &Helper,
     _: &Handlebars,
     _: &Context,
     _: &mut RenderContext,
     out: &mut dyn Output,
 ) -> HelperResult {
-    out.write(&util::reverse_domain(get_str(helper)))
-        .map_err(Into::into)
+    let last = get_str(helper).split('.').rev().next().unwrap_or_default();
+    out.write(&last).map_err(Into::into)
 }
 
-fn reverse_domain_snake_case(
+fn ident_no_last_part(
     helper: &Helper,
     _: &Handlebars,
     _: &Context,
     _: &mut RenderContext,
     out: &mut dyn Output,
 ) -> HelperResult {
-    use heck::ToSnekCase as _;
-    out.write(&util::reverse_domain(get_str(helper)).to_snek_case())
-        .map_err(Into::into)
+    let components = get_str(helper).split('.').collect::<Vec<_>>();
+    let ident_no_last = components[..components.len() - 1].join(".");
+    out.write(&ident_no_last).map_err(Into::into)
 }
 
 fn escape_kotlin_keyword(
@@ -237,11 +237,8 @@ pub fn init(config: Option<&Config>) -> Bicycle {
                 Box::new(quote_and_join_colon_prefix),
             );
             helpers.insert("snake-case", Box::new(snake_case));
-            helpers.insert("reverse-domain", Box::new(reverse_domain));
-            helpers.insert(
-                "reverse-domain-snake-case",
-                Box::new(reverse_domain_snake_case),
-            );
+            helpers.insert("ident-no-last-part", Box::new(ident_no_last_part));
+            helpers.insert("ident-last-part", Box::new(ident_last_part));
             helpers.insert("escape-kotlin-keyword", Box::new(escape_kotlin_keyword));
             helpers.insert("dot-to-slash", Box::new(dot_to_slash));
             if config.is_some() {

--- a/templates/apps/dioxus/Cargo.toml.hbs
+++ b/templates/apps/dioxus/Cargo.toml.hbs
@@ -12,7 +12,7 @@ name = "{{app.name}}-desktop"
 path = "gen/bin/desktop.rs"
 
 [package.metadata.cargo-android]
-app-activity-name = "{{reverse-domain app.identifier}}.MainActivity"
+app-activity-name = "{{app.identifier}}.MainActivity"
 app-dependencies = [
     "androidx.webkit:webkit:1.6.1",
     "androidx.appcompat:appcompat:1.6.1",
@@ -25,9 +25,9 @@ app-theme-parent = "Theme.MaterialComponents.DayNight.DarkActionBar"
 vulkan-validation = false
 
 [package.metadata.cargo-android.env-vars]
-WRY_ANDROID_PACKAGE = "{{reverse-domain app.identifier}}"
+WRY_ANDROID_PACKAGE = "{{app.identifier}}"
 WRY_ANDROID_LIBRARY = "{{snake-case app.name}}"
-WRY_ANDROID_KOTLIN_FILES_OUT_DIR = "<android-project-dir>/app/src/main/kotlin/{{dot-to-slash (reverse-domain app.identifier)}}"
+WRY_ANDROID_KOTLIN_FILES_OUT_DIR = "<android-project-dir>/app/src/main/kotlin/{{dot-to-slash app.identifier}}"
 
 [package.metadata.cargo-apple.ios]
 frameworks = [ "WebKit" ]

--- a/templates/apps/dioxus/gen/android/app/src/main/kotlin/{{dot-to-slash (reverse-domain app.domain)}}/{{snake-case app.name}}/MainActivity.kt.hbs
+++ b/templates/apps/dioxus/gen/android/app/src/main/kotlin/{{dot-to-slash (reverse-domain app.domain)}}/{{snake-case app.name}}/MainActivity.kt.hbs
@@ -1,3 +1,0 @@
-package {{escape-kotlin-keyword (reverse-domain app.identifier)}}
-
-class MainActivity : WryActivity()

--- a/templates/apps/dioxus/gen/android/app/src/main/kotlin/{{dot-to-slash app.identifier}}/MainActivity.kt.hbs
+++ b/templates/apps/dioxus/gen/android/app/src/main/kotlin/{{dot-to-slash app.identifier}}/MainActivity.kt.hbs
@@ -1,0 +1,3 @@
+package {{escape-kotlin-keyword app.identifier}}
+
+class MainActivity : WryActivity()

--- a/templates/apps/dioxus/src/lib.rs.hbs
+++ b/templates/apps/dioxus/src/lib.rs.hbs
@@ -37,12 +37,13 @@ pub extern "C" fn start_app() {
     #[cfg(target_os = "android")]
     {
         tao::android_binding!(
-            {{reverse-domain-snake-case app.identifier}},
+            {{snake-case (ident-no-last-part app.identifier)}},
+            {{snake-case (ident-last-part app.name)}},
             WryActivity,
             wry::android_setup, // pass the wry::android_setup function to tao which will invoke when the event loop is created
             _start_app
         );
-        wry::android_binding!(com_example, {{snake-case app.name}});
+        wry::android_binding!({{snake-case (ident-no-last-part app.identifier)}}, {{snake-case (ident-last-part app.name)}});
     }
 
     #[cfg(target_os = "ios")]

--- a/templates/apps/wry/Cargo.toml.hbs
+++ b/templates/apps/wry/Cargo.toml.hbs
@@ -12,7 +12,7 @@ name = "{{app.name}}-desktop"
 path = "gen/bin/desktop.rs"
 
 [package.metadata.cargo-android]
-app-activity-name = "{{reverse-domain app.identifier}}.MainActivity"
+app-activity-name = "{{app.identifier}}.MainActivity"
 app-dependencies = [
     "androidx.webkit:webkit:1.6.1",
     "androidx.appcompat:appcompat:1.6.1",
@@ -25,9 +25,9 @@ app-theme-parent = "Theme.MaterialComponents.DayNight.DarkActionBar"
 vulkan-validation = false
 
 [package.metadata.cargo-android.env-vars]
-WRY_ANDROID_PACKAGE = "{{reverse-domain app.identifier}}"
+WRY_ANDROID_PACKAGE = "{{app.identifier}}"
 WRY_ANDROID_LIBRARY = "{{snake-case app.name}}"
-WRY_ANDROID_KOTLIN_FILES_OUT_DIR = "<android-project-dir>/app/src/main/kotlin/{{dot-to-slash (reverse-domain app.identifier)}}"
+WRY_ANDROID_KOTLIN_FILES_OUT_DIR = "<android-project-dir>/app/src/main/kotlin/{{dot-to-slash app.identifier}}"
 
 [package.metadata.cargo-apple.ios]
 frameworks = [ "WebKit" ]

--- a/templates/apps/wry/gen/android/app/src/main/kotlin/{{dot-to-slash (reverse-domain app.domain)}}/{{snake-case app.name}}/MainActivity.kt.hbs
+++ b/templates/apps/wry/gen/android/app/src/main/kotlin/{{dot-to-slash (reverse-domain app.domain)}}/{{snake-case app.name}}/MainActivity.kt.hbs
@@ -1,3 +1,0 @@
-package {{escape-kotlin-keyword (reverse-domain app.identifier)}}
-
-class MainActivity : WryActivity()

--- a/templates/apps/wry/gen/android/app/src/main/kotlin/{{dot-to-slash app.identifier}}/MainActivity.kt.hbs
+++ b/templates/apps/wry/gen/android/app/src/main/kotlin/{{dot-to-slash app.identifier}}/MainActivity.kt.hbs
@@ -1,0 +1,3 @@
+package {{escape-kotlin-keyword app.identifier}}
+
+class MainActivity : WryActivity()

--- a/templates/apps/wry/src/lib.rs.hbs
+++ b/templates/apps/wry/src/lib.rs.hbs
@@ -42,12 +42,13 @@ pub extern "C" fn start_app() {
     #[cfg(target_os = "android")]
     {
         tao::android_binding!(
-            {{reverse-domain-snake-case app.identifier}},
+            {{snake-case (ident-no-last-part app.identifier)}},
+            {{snake-case (ident-last-part app.name)}},
             WryActivity,
             wry::android_setup, // pass the wry::android_setup function to tao which will invoke when the event loop is created
             _start_app
         );
-        wry::android_binding!(com_example, {{snake-case app.name}});
+        wry::android_binding!({{snake-case (ident-no-last-part app.identifier)}}, {{snake-case (ident-last-part app.name)}});
     }
 
     #[cfg(target_os = "ios")]

--- a/templates/platforms/android-studio/app/build.gradle.kts.hbs
+++ b/templates/platforms/android-studio/app/build.gradle.kts.hbs
@@ -5,11 +5,11 @@ plugins {
 }
 
 android {
-    namespace="{{reverse-domain app.identifier}}"{{#if has-asset-packs}}
+    namespace="{{app.identifier}}"{{#if has-asset-packs}}
     assetPacks += mutableSetOf({{quote-and-join-colon-prefix asset-packs}}){{/if}}
     compileSdk = 33
     defaultConfig {
-        applicationId = "{{reverse-domain app.identifier}}"
+        applicationId = "{{app.identifier}}"
         minSdk = {{android.min-sdk-version}}
         targetSdk = 33
         versionCode = 1

--- a/templates/platforms/xcode/project.yml.hbs
+++ b/templates/platforms/xcode/project.yml.hbs
@@ -1,6 +1,6 @@
 name: {{app.name}}
 options:
-  bundleIdPrefix: {{reverse-domain app.identifier}}
+  bundleIdPrefix: {{app.identifier}}
   deploymentTarget:
     iOS: {{apple.ios-version}}
     macOS: {{apple.macos-version}}
@@ -12,7 +12,7 @@ settingGroups:
   app:
     base:
       PRODUCT_NAME: {{app.stylized-name}}
-      PRODUCT_BUNDLE_IDENTIFIER: {{reverse-domain app.identifier}}
+      PRODUCT_BUNDLE_IDENTIFIER: {{app.identifier}}
       {{#if apple.development-team}}
       DEVELOPMENT_TEAM: {{apple.development-team}}
       {{/if}}

